### PR TITLE
MEPTS-1442 | AVALIACAO DE MDS CHANGES

### PR DESCRIPTION
**Pull request by helderjosue**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListOfPatientsWithMdsEvaluationCohortDataset.java [hunk 1]
```diff
@@ -303,24 +304,26 @@
...
         endDateMappings,
         new GeneralDateConverter());
 
-    // C.2 - Data de registo do resultado da CV de Seguimento - Sheet 1: (coluna AQ)
+    // C.2 - Data de registo do resultado da CV entre 12º e 24º mês do TARV: (coluna AQ)
     pdd.addColumn(
         "cv_result_date_c",
-        listOfPatientsWithMdsEvaluationCohortQueries.getSecondViralLoadResultDate(),
+        listOfPatientsWithMdsEvaluationCohortQueries
+            .getLastViralLoadResultDateBetweenPeriodsInMonthsAfterTarv(12, 36),
```
Comment (helderjosue) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListOfPatientsWithMdsEvaluationCohortDataset.java]: Please correct the number of months

Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListOfPatientsWithMdsEvaluationCohortDataset.java]: Done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsWithMdsEvaluationCohortQueries.java [hunk 2]
```diff
@@ -1121,30 +1123,33 @@ public DataDefinition getSecondViralLoadResult() {
...
-            + "  ) second_vl_date "
-            + " ON second_vl_date.patient_id = second_vl.patient_id "
-            + "WHERE second_vl.voided = 0 "
-            + "AND ee.voided = 0 "
-            + "AND oo.voided = 0 "
-            + "AND ee.encounter_datetime = second_vl_date.second_vl_date "
-            + "AND      ee.encounter_type = ${6} "
-            + "AND        ee.location_id = :location "
+        "SELECT     p.patient_id, "
+            + "     IF(o.value_numeric IS NOT NULL, o.value_numeric, IF(o.value_coded = 165331, CONCAT('MENOR QUE ',o.comments), o.value_coded)) AS first_vl_result  "
```
Comment (helderjosue) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsWithMdsEvaluationCohortQueries.java]: This if clause `IF(o.value_numeric IS NOT NULL, o.value_numeric, ...` might not always return the expected result, due to the` o.value_numeric IS NOT NULL` verification that can return true but for a different concept.
Ex: if a patient is included on the definition with qualitative VL (`o.concept_id = ${1305} and o.value_coded IS NOT NULL`), when the IF clause is triggered, if the patient has another **concept with value_numeric not null**, the **o.value_numeric** (which is not the criteria including him), can return true. 

So, you have to make sure that the returned results are exactly from the criteria that is including the patient on the definition.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListOfPatientsWithMdsEvaluationCohortDataset.java
Comment (helderjosue) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListOfPatientsWithMdsEvaluationCohortDataset.java:1]: Check the implementation of the method - `getCd4ResultSectionC(12,24)` column AS
The result returned is not from the latest cd4 registered on the given period 
check [https://github.com/jembi/openmrs-module-eptsreports/blob/92b72e16703b561cbf3b2ff538ff371d59ae0de5/api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsWithMdsEvaluationCohortQueries.java#L1324](url)

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsWithMdsEvaluationCohortQueries.java [hunk 3]
```diff
@@ -1121,30 +1123,33 @@
...
```
Comment (helderjosue) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsWithMdsEvaluationCohortQueries.java:1127]: Please follow the same approach reported for VL cases

**Review by helderjosue — COMMENTED** [review]
please check the comments

**Review by helderjosue — COMMENTED** [review]
Looks good to me! Therefore, I cant approve this PR, it was created by me.

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**